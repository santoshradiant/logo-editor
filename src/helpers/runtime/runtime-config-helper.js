let runTimeConfig

// modify process env to new style
if (!runTimeConfig) {
  runTimeConfig = {}
}

function setRuntimeConfig (config) {
  runTimeConfig = config
}

function getRuntimeConfig () {
  return runTimeConfig
}

const appGateway = {
  development: 'https://api-gw.uat.builderservices.io',
  uat: 'https://api-gw.uat.builderservices.io',
  qa: 'https://api-gw.qa.builderservices.io',
  production: 'https://api-gw.builderservices.io'
}

const runtime = {
  _defaultApiName: 'logobuilder-api',
  version: 2,
  _brand: 'websitebuilder',
  _mfe: true,
  _fetchXsrfTokenHeaderName: 'X-CSRF-TOKEN',
  _env: process.env.NODE_ENV,
  _settings: { 'logomaker-brandname-readme-lock': false },
  _brandedUrls: {
    app_gateway: {
      api: appGateway[process.env.NODE_ENV]
    }
  }
}

const createRuntime = env => {
  return {
    ...runtime
  }
}

export { setRuntimeConfig, getRuntimeConfig, runtime, createRuntime }
