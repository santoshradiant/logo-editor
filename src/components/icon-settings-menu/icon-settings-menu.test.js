import 'jest-styled-components'
import 'core-js/stable'
import 'regenerator-runtime/runtime'
import React from 'react'
import userEvent from '@testing-library/user-event'

import { render, within, screen } from 'test-wrapper'

import IconSettingsMenu from './index'
import MockSettings from './__mock__/index'

test('Displays icon-settings-menu', async () => {
  const selected = MockSettings[0].name
  const setSelected = jest.fn()

  const { container } = await render(
    <IconSettingsMenu items={MockSettings} activeName={selected} setActive={setSelected} />
  )
  // Compare snapshot
  expect(container).toMatchSnapshot()

  // check if the tab is selected
  const activeElement = screen.getAllByRole('tab', { selected: true })[0]
  const activeLabel = within(activeElement).getByTestId('label')
  expect(activeLabel).toHaveTextContent(selected)

  // Fire click event and check that
  const nonSelectedItem = screen.getAllByRole('tab', { selected: false })[1]
  await userEvent.click(nonSelectedItem)
  expect(setSelected).toBeCalled()
})

test('Icon-settings-menu has focus', () => {
  const selected = MockSettings[0].name
  const setSelected = jest.fn()
  render(<IconSettingsMenu items={MockSettings} activeName={selected} setActive={setSelected} />)
  const [first, second] = screen.getAllByRole('tab')
  expect(document.body).toHaveFocus()
  userEvent.tab()
  expect(first).toHaveFocus()
  userEvent.tab()
  expect(second).toHaveFocus()
})
