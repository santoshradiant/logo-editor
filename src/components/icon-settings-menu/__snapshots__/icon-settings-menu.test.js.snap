// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays icon-settings-menu 1`] = `
<div>
  <div
    class="app-logo"
  >
    <div
      class="app-logo-NotificationBar-fullScreenVariant-1"
    />
    <div
      class="icon-settings-menuview__FullHeightWrapper-sifysq-4 jTCuVw"
    >
      <div
        class="icon-settings-menuview__IconContainer-sifysq-2 hYgNBi"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="icon-settings-menuview__IconBarButton-sifysq-3 cfISAO"
          role="tab"
        >
          <div
            class="app-logo-MuiGrid-root icon-settings-menuview__StyledGrid-sifysq-0 gatdKn app-logo-MuiGrid-container app-logo-MuiGrid-direction-xs-column app-logo-MuiGrid-align-items-xs-center app-logo-MuiGrid-justify-xs-center"
          >
            <svg
              aria-hidden="true"
              class="app-logo-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="icon-settings-menuview__Label-sifysq-1 ROzOT"
              data-testid="label"
            >
              test1
            </span>
          </div>
        </button>
        <button
          aria-selected="false"
          class="icon-settings-menuview__IconBarButton-sifysq-3 fkXDUo"
          role="tab"
        >
          <div
            class="app-logo-MuiGrid-root icon-settings-menuview__StyledGrid-sifysq-0 gatdKn app-logo-MuiGrid-container app-logo-MuiGrid-direction-xs-column app-logo-MuiGrid-align-items-xs-center app-logo-MuiGrid-justify-xs-center"
          >
            <svg
              aria-hidden="true"
              class="app-logo-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="icon-settings-menuview__Label-sifysq-1 ROzOT"
              data-testid="label"
            >
              test2
            </span>
          </div>
        </button>
        <button
          aria-selected="false"
          class="icon-settings-menuview__IconBarButton-sifysq-3 fkXDUo"
          role="tab"
        >
          <div
            class="app-logo-MuiGrid-root icon-settings-menuview__StyledGrid-sifysq-0 gatdKn app-logo-MuiGrid-container app-logo-MuiGrid-direction-xs-column app-logo-MuiGrid-align-items-xs-center app-logo-MuiGrid-justify-xs-center"
          >
            <svg
              aria-hidden="true"
              class="app-logo-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="icon-settings-menuview__Label-sifysq-1 ROzOT"
              data-testid="label"
            >
              test3
            </span>
          </div>
        </button>
        <button
          aria-selected="false"
          class="icon-settings-menuview__IconBarButton-sifysq-3 fkXDUo"
          role="tab"
        >
          <div
            class="app-logo-MuiGrid-root icon-settings-menuview__StyledGrid-sifysq-0 gatdKn app-logo-MuiGrid-container app-logo-MuiGrid-direction-xs-column app-logo-MuiGrid-align-items-xs-center app-logo-MuiGrid-justify-xs-center"
          >
            <svg
              aria-hidden="true"
              class="app-logo-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="icon-settings-menuview__Label-sifysq-1 ROzOT"
              data-testid="label"
            >
              test4
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
